static/src/js/components/OrderOptionList/OrderOptionList.jsx

```diff
@@ -0,0 +1,220 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState
+} from 'react'
+import { useLazyQuery } from '@apollo/client'
+import {
+  Col,
+  Placeholder,
+  Row
+} from 'react-bootstrap'
+import { useSearchParams } from 'react-router-dom'
+import moment from 'moment'
+
+import ControlledPaginatedContent from '../ControlledPaginatedContent/ControlledPaginatedContent'
+import EllipsisLink from '../EllipsisLink/EllipsisLink'
+import ErrorBanner from '../ErrorBanner/ErrorBanner'
+import Page from '../Page/Page'
+import Table from '../Table/Table'
+
+import useAppContext from '../../hooks/useAppContext'
+
+import { GET_ORDER_OPTIONS } from '../../operations/queries/getOrderOptions'
+
+import errorLogger from '../../utils/errorLogger'
+import parseError from '../../utils/parseError'
+
+const OrderOptionList = () => {
+  const { user } = useAppContext()
+  const { providerId } = user
+  const [searchParams, setSearchParams] = useSearchParams()
+
+  const [orderOptionList, setOrderOptionList] = useState()
+  const [loading, setLoading] = useState()
+  const [error, setError] = useState()
+
+  const activePage = parseInt(searchParams.get('page'), 10) || 1
+
+  const limit = 20
+  const offset = (activePage - 1) * limit
+
+  const setPage = (nextPage) => {
+    setSearchParams((currentParams) => {
+      currentParams.set('page', nextPage)
+
+      return {
+        ...Object.fromEntries(currentParams)
```

Comment:
This doesn't need to be an object, you can just return `Object.fromEntries(currentParams)`